README.md

```diff
@@ -1,2 +1,73 @@
-# declarative-custom-elements
-Explainer draft for specifying an approach to Declarative Custom Elements
+# Declarative Custom Elements
+### A discussion...
+
+## Problem
+Imperative JS code is required to unlock the power of [Custom Elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements), [Shadow DOM](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM), and other "web components" APIs. This is particularly true when these APIs are used in concert with each other. A declarative API would open the door for things like:
+
+- non-JS developers being given the power of custom elements
+- having custom element definitions available as parse time without requireing the jump over to the JS thread early in the document parse
+- others?
+
+*Questions:* Have we come to a future where "web components" amount to _just_ Custom Elements and Shadow DOM? Could we benefit from including other specs in this conversation? Possibly:
+
+- [constructible stylesheets](https://web.dev/constructable-stylesheets/)
+- [adoptedStyleSheets](https://wicg.github.io/construct-stylesheets/#using-constructed-stylesheets)
+- [aria delegation](https://leobalter.github.io/cross-root-aria-delegation/) (possibly implied but also not currently "spec", only proposal)
+- [scoped registries](https://github.com/WICG/webcomponents/issues/716)
+- others? see [current Web Components Community Group priorities](https://w3c.github.io/webcomponents-cg/)
```

Comment:
listed features are needed for the concept of mature DCE libs, but even partial implementation of stack is quite powerful.

To make the maturity level comparable to current JS stack still needed:
- lib level scope (namespace) for CSS/JS/importmaps  
- scope definitions which would allow to override CSS internal rules. Do not confuse with complete "html inlude" proposal.
- module loader. The DCE need ability to be loaded as module with all matching perks:
    * relative path to internally referenced resources. Otherwise all urls would be resolved relatively page [baseURI](https://developer.mozilla.org/en-US/docs/Web/API/Node/baseURI)
    *  importmaps to be controlled by app AND sub-module, Current JS-centric implementation has a gap there.

Looking into future:
- Declarative DAL/data island/Data binding. Including transformation of event source data to targets.
- ability to insulate and override JS VM as in [microapplication](https://cdn.xml4jquery.com/ajax/libs/embed-page/0.0.20/build/esm-unbundled/demo/current-script.html). Scoped registries and importmaps fits there naturally.